scriptmodules/emulators/mame4all.sh

```diff
@@ -22,7 +22,11 @@ function sources_mame4all() {
 
 function build_mame4all() {
     make clean
-    make
+    if isPlatform "rpi2"; then
+        make PLATFORM=rpi2 -j 4
+    else
+        make PLATFORM=rpi1
+    fi
     md_ret_require="$md_build/mame"
```

Comment:
see https://github.com/RetroPie/RetroPie-Setup/blob/master/scriptmodules/system.sh - you can build on all 4 cores by using MAKEFLAGS as needed. this also handles the compiler flags for platform.